pytorch_lightning/loops/fit_loop.py

```diff
@@ -0,0 +1,300 @@
+        return self.training_loop.min_steps
+
+    @property
+    def max_steps(self) -> int:
+        """Returns the maximum number of steps to run"""
+        return self.training_loop.max_steps
+
+    @max_steps.setter
+    def max_steps(self, value: int) -> None:
+        """Sets the maximum number of steps (forwards to training_loop)"""
+        # TODO(@awaelchli): This setter is required by debugging connector (fast dev run), should be avoided
+        self.training_loop.max_steps = value
+
+    @property
+    def running_loss(self) -> TensorRunningAccum:
+        """Returns the running loss"""
+        return self.training_loop.batch_loop.running_loss
+
+    @property
+    def _skip_backward(self) -> bool:
+        """ Determines whether the loop will skip backward during automatic optimization. """
+        return self.training_loop.batch_loop._skip_backward
+
+    @_skip_backward.setter
+    def _skip_backward(self, value: bool) -> None:
+        """ Determines whether the loop will skip backward during automatic optimization. """
+        self.training_loop.batch_loop._skip_backward = value
+
+    @property
+    def done(self) -> bool:
+        """Evaluates when to leave the loop.
+
+        Returns True if trainer.should_stop was set (e.g. by early stopping)
+        or if the maximum number of steps or epochs is reached.
+        """
+        # TODO(@awaelchli): Move track steps inside training loop and move part of these condition inside training loop
+        stop_steps = self.max_steps is not None and self.global_step >= self.max_steps
+        stop_epochs = self.max_epochs is not None and self.current_epoch >= self.max_epochs
+
+        should_stop = False
+        if self.trainer.should_stop:
+            # early stopping
+            met_min_epochs = self.current_epoch >= self.min_epochs if self.min_epochs else True
+            met_min_steps = self.global_step >= self.min_steps if self.min_steps else True
+            if met_min_epochs and met_min_steps:
+                should_stop = True
+            else:
+                log.info(
+                    'Trainer was signaled to stop but required minimum epochs'
+                    f' ({self.min_epochs}) or minimum steps ({self.min_steps}) has'
+                    ' not been met. Training will continue...'
+                )
+                self.trainer.should_stop = False
+
+        return stop_steps or should_stop or stop_epochs
+
+    def connect(self, trainer: 'pl.Trainer', *args: Any, **kwargs: Any) -> None:
+        """Connects the loop with necessary arguments like the trainer"""
+        # TODO(@justusschock): Do we want to forward *args and **kwargs to the inner loop here?
+        # TODO(@justusschock): Can we make the trainer a weakref/proxy?
+        void(*args, **kwargs)
+        self.trainer = trainer
+        self.training_loop.connect(trainer)
+
+    def reset(self) -> None:
+        """Resets the internal state of this loop"""
```

Comment:
Since this is empty, should the parent method not be abstract?

Reply:
I think it should still be abstract. This just means that users should be aware that if they don't implement it/have it empty the state will be reused which is likely to introduce undesired behavour. So if you really implement it like this, it is an active choice and you are aware of the consequences.